src/tuesday/core/plotting/powerspectra.py

```diff
@@ -0,0 +1,229 @@
+"""Plotting functions for 1D and 2D power spectra."""
+
+import astropy.units as un
+import matplotlib.pyplot as plt
+import numpy as np
+from astropy.cosmology.units import littleh
+from matplotlib import rcParams
+
+
+def plot_1d_power_spectrum(
+    wavemodes: un.Quantity,
+    power_spectrum: un.Quantity,
+    title: str | None = None,
+    xlabel: str | None = None,
+    ylabel: str | None = None,
+    colors: list | None = None,
+    fontsize: float | None = 16,
+    save_path: str | None = None,
+) -> None:
+    """
+    Plot a 1D power spectrum.
+
+    Parameters
+    ----------
+    wavemodes : np.ndarray
+        Wavemodes corresponding to the power spectrum.
+    power_spectrum : np.ndarray
+        Power spectrum values.
+    title : str, optional
+        Title of the plot.
+    xlabel : str, optional
+        Label for the x-axis.
+    ylabel : str, optional
+        Label for the y-axis.
+    colors : list, optional
+        List of colors for each line in the plot.
+    save_path : str, optional
+        Path to save the plot. If None, the plot will be shown instead.
+    """
+    plt.figure(figsize=(8, 6))
+    rcParams.update({"font.size": fontsize})
+    n = power_spectrum.shape[0]
+    if colors is None:
+        colors = ["k"] * n
+    if xlabel is None:
+        if wavemodes.unit == 1 / un.Mpc:
+            xlabel = r"$k \, [\rm{Mpc}^{-1}]$"
+        elif wavemodes.unit == littleh / un.Mpc:
+            xlabel = r"$k \, [h \, \rm{Mpc}^{-1}]$"
+        else:
+            raise ValueError("Wavemodes must be in units of 1/Mpc or h/Mpc.")
+    if ylabel is None:
+        if power_spectrum.unit == un.mK**2 * un.Mpc**3:
+            ylabel = r"$P(k) \, [\rm{Mpc}^{3}]$"
+        elif power_spectrum.unit == un.mK**2 * un.Mpc**3 / littleh**3:
+            ylabel = r"$P(k) \, [h^{-3} \rm{Mpc}^{3}]$"
+        elif power_spectrum.unit == un.mK**2:
+            ylabel = r"$\Delta^2_{21} [\rm{mK}^2]$"
+        else:
+            raise ValueError(
+                "Accepted power spectrum units: mK^2 Mpc^3, mK^2 Mpc^3/h^3 or mK^2."
+            )
+    for i in range(n):
+        plt.plot(wavemodes, power_spectrum[i], color=colors[i])
+    if title is not None:
+        plt.title(title, fontsize=fontsize)
+    plt.xlabel(xlabel, fontsize=fontsize)
+    plt.ylabel(ylabel, fontsize=fontsize)
+    if np.all(np.diff(wavemodes) != np.diff(wavemodes)[0]):
+        plt.loglog()
+    else:
+        plt.yscale("log")
+
+    if save_path is not None:
+        plt.savefig(save_path)
+        plt.close()
+    else:
+        plt.show()
```

Comment:
Let's not save/show plots in functions. Let the user do that in whatever code they have. Because there are all kinds of options you might want to pass to the `savefig` call as well.